infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/KubernetesPluginsToolingApplier.java

```diff
@@ -41,14 +42,21 @@
 public class KubernetesPluginsToolingApplier implements ChePluginsApplier {
 
   private final String defaultSidecarMemoryLimitBytes;
+  private final String sidecarImagePullPolicy;
   private final boolean isAuthEnabled;
 
   @Inject
   public KubernetesPluginsToolingApplier(
+      @Named("che.workspace.sidecar.image_pull_policy") String sidecarImagePullPolicy,
       @Named("che.workspace.sidecar.default_memory_limit_mb") long defaultSidecarMemoryLimitMB,
       @Named("che.agents.auth_enabled") boolean isAuthEnabled) {
     this.defaultSidecarMemoryLimitBytes = String.valueOf(defaultSidecarMemoryLimitMB * 1024 * 1024);
     this.isAuthEnabled = isAuthEnabled;
+    this.sidecarImagePullPolicy =
```

Comment:
Don't you think that set in a static field will be simpler?

Reply:
It also makes sense to consider adding @Singleton annotation here since a constructor does some work and may take some time.

Reply:
Added